### ISSUE TITLE
ProcessorListener should increase DEFAULT_QUEUE_CAPACITY

### ISSUE THREAD

**Issue by cizezsy** [kubernetes-client/java#667]
The [default capacity](https://github.com/kubernetes-client/java/blob/master/util/src/main/java/io/kubernetes/client/informer/cache/ProcessorListener.java#L20) of queue in ProcessorListener is 1000.

But it will easily cause IllegalStateExecption and makes the controller thread dead when we use informer to watch more than a thousand resources.

Is there any possibility to increase the number? I know it may cause OOM when processListener is stalled or blocked, but I think the current number is too small....Or, at least, could we make this configurable?

Stacktrace:
```
java.lang.IllegalStateException: Queue full
	at java.util.AbstractQueue.add(AbstractQueue.java:98)
	at java.util.concurrent.ArrayBlockingQueue.add(ArrayBlockingQueue.java:312)
	at io.kubernetes.client.informer.cache.ProcessorListener.add(ProcessorListener.java:75)
	at io.kubernetes.client.informer.cache.SharedProcessor.distribute(SharedProcessor.java:97)
	at io.kubernetes.client.informer.impl.DefaultSharedIndexInformer.handleDeltas(DefaultSharedIndexInformer.java:187)
	at io.kubernetes.client.informer.cache.DeltaFIFO.pop(DeltaFIFO.java:313)
	at io.kubernetes.client.informer.cache.Controller.processLoop(Controller.java:140)
	at io.kubernetes.client.informer.cache.Controller.run(Controller.java:107)
	at java.lang.Thread.run(Thread.java:748)

```

**Comment by yue9944882** [comment]
i try to fix it in other way in https://github.com/kubernetes-client/java/pull/666/commits/b18dbfdc9820a9c575c114ce68ffd30359bcf91b. we use add instead of offer, the latter won't throw exception if the queue is full. how do you think about that?

**Comment by cizezsy** [comment]
@yue9944882 Thanks! The change looks great.

But I do not think discarding the event when the queue is full is a good choice. Is there any special reason to do that? It looks pretty strange to users that some events dropped unexpectedly. Maybe it is better to replace the ArrayBlockingQueue with LinkedBlockQueue to remove the capacity limit.

**Comment by yue9944882** [comment]
> Maybe it is better to replace the ArrayBlockingQueue with LinkedBlockQueue to remove the capacity limit.

i think you're right, how do you think about sending a fix?

the previous fix is merely defensive. for bounding the queue's memory usage, we have to either drop or block the writes.

**Comment by cizezsy** [comment]
@yue9944882 sent a fix #669 for this, thanks~
